src/ResourceManager/Common/Commands.ResourceManager.Common/ArgumentCompleters/PSCompleterBase.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System;
+
+namespace Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters
+{
+    /// <summary>
+    /// This attribute will allow the user to autocomplete the -Location parameter of a cmdlet with valid locations (as determined by the list of ResourceTypes given)
+    /// </summary>
+    public class PSCompleterBaseAttribute : Attribute
+    {
+        public PSCompleterBaseAttribute(params string[] arguments)
+        {
+        }
+
+        public virtual string[] GetCompleterValues()
```

Comment:
You can make this abstract, so that no impementation is necessary (If the class is abstract, it also cannot be created).

Reply:
done